qemu/tests/vpmu_check_instructions.py

```diff
@@ -0,0 +1,67 @@
+import os
+import logging
+import re
+
+from shutil import copyfile
+
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import error_context
+from virttest import arch
+from virttest import utils_package
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    vpum cpu cycles checking between host and guest:
+    1) boot guest
+    2) check cpu cycles for host
+    3) check cpu cycles for guest and compare with host
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    check_timeout = int(params.get("check_timeout", "600"))
+    tmp_dir = params.get("tmp_dir")
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = params.get_numeric("login_timeout", 360)
+    session = vm.wait_for_login(timeout=timeout)
+    test_cmd = params.get("test_cmd")
+    build_cmd = params.get("build_cmd")
+    vm_arch = params["vm_arch_name"]
+    host_arch = arch.ARCH
+    src_dir = os.path.join(data_dir.get_deps_dir(), 'million')
+    src_file = os.path.join(src_dir, "million-%s.s" % host_arch)
+    dst_file = os.path.join(tmp_dir, "million-%s.s" % host_arch)
+    utils_package.package_install("perf")
+
+    error_context.context("build binary file 'million' in host", logging.info)
+    copyfile(src_file, dst_file)
+    process.run(build_cmd % host_arch)
+
+    error_context.context("running test command in host", logging.info)
+    s, o = process.getstatusoutput(test_cmd)
+    if s:
+        test.fail("Failed to run test command")
+
+    host_cpu_cycles = re.findall(r"(\d+) *instructions:u", o, re.M)
+
+    utils_package.package_install("perf", session)
```

Comment:
In my opinions, we need to check the installation status
if not utils_package.package_install(pkgs, session):
test.error("Install dependency packages failed")

Reply:
Also has the following problem while applying it 
[root@ibm-p9b-12 io-github-autotest-qemu]# git am 2678.patch
Applying: vpmu:cpu cycles checking between host and guest
.git/rebase-apply/patch:21: space before tab in indent.
     	     # count for 1 million instructions
.git/rebase-apply/patch:51: space before tab in indent.
     	     # count for 1 million instructions
warning: 2 lines add whitespace errors.